react/ShelfItem.js

```diff
@@ -27,7 +27,7 @@ export default class ShelfItem extends Component {
 
   render() {
     const { item, summary } = this.props
-    summary.name.tag = 'h2'
+    assocPath(['name', 'tag'], 'h2', summary)
```

Comment:
```suggestion
    const newSummary = assocPath(['name', 'tag'], 'h2', summary)
```

Reply:
Line 30 was not actually doing anything, since Ramda operates without side effects.

My suggestion only works if you pass `{...newSummary` instead of `{... sumary}` for the child component